src/main/java/graphql/schema/idl/TypeDefinitionRegistry.java

```diff
@@ -67,6 +70,12 @@ public TypeDefinitionRegistry merge(TypeDefinitionRegistry typeRegistry) throws
             defined.ifPresent(errors::add);
         });
 
+        Map<String, DirectiveDefinition> tempDirectiveDefs = new LinkedHashMap<>();
```

Comment:
Thanks for working on this.

It looks like `tempDirectiveDefs` is never used to update the registry further down in the merge method:

```java
        // ok commit to the merge
        this.types.putAll(tempTypes);
        this.scalarTypes.putAll(tempScalarTypes);
        this.directiveDefinitions.putAll(tempDirectiveDefs); // <- this statement appears to be missing
```